src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,23 +34,28 @@ export const TodoFilter = () => (
 
     <p className="control is-expanded has-icons-left has-icons-right">
       <input
+        value={query}
         data-cy="searchInput"
         type="text"
         className="input"
         placeholder="Search..."
+        onChange={(event) => onQuery(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
+      {query && (
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
```

Comment:
Try to fix problems, not disable eslint
[eslint-disable-next-line jsx-a11y/control-has-associated-label](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/label-has-associated-control.md)